lib/bolt/outputter/human.rb

```diff
@@ -311,13 +311,16 @@ def print_module_list(module_list)
         end
       end
 
-      def print_targets(options)
-        targets = options[:targets].map(&:name)
+      def print_targets(targets)
         count = "#{targets.count} target#{'s' unless targets.count == 1}"
-        @stream.puts targets.join("\n")
+        @stream.puts targets.map(&:name).join("\n")
         @stream.puts colorize(:green, count)
       end
 
+      def print_target_info(targets)
+        @stream.puts targets.map(&:detail).to_yaml
+      end
```

Comment:
The Psych `to_yaml` does a super annoying thing where it tries to create and refence anchors for common data. This means that the output of this, if you have multiple targets, is:
```yaml
- name: hostname1
  uri: hostname1
  alias: 
  config:
    transport: ssh
    ssh:
      connect-timeout: 10
      tty: false
      load-config: true
      disconnect-timeout: 5
      host-key-check: false
  vars: {}
  features: []
  facts: {}
  plugin_hooks:
    puppet_library: &1
      plugin: install_agent
- name: hostname2
  uri: hostname2
  alias: 
  config:
    transport: ssh
    ssh:
      connect-timeout: 10
      tty: false
      load-config: true
      disconnect-timeout: 5
      host-key-check: false
  vars: {}
  features: []
  facts: {}
  plugin_hooks:
    puppet_library: *1
- name: hostname3
  uri: hostname3
  alias: 
  config:
    transport: ssh
    ssh:
      connect-timeout: 10
      tty: false
      load-config: true
      disconnect-timeout: 5
      host-key-check: false
  vars: {}
  features: []
  facts: {}
  plugin_hooks:
    puppet_library: *1
```

If you directly copy paste this into `inventory.yaml`, you get:
```
NoMethodError: undefined method `line' for #<Psych::BadAlias: Unknown alias: 1>
```

In looking into Psych a bit I'm not sure there's a way to disable or configure aliasing. Is there a workaround for this?

Reply:
The second comment here perhaps? https://stackoverflow.com/questions/3981128/ruby-yaml-write-without-aliases

Reply:
Fixed with a deep clone of the plugins hash.